src/constraint/spec.ts

```diff
@@ -51,6 +51,18 @@ export class SpecConstraintModel extends AbstractConstraintModel {
               return some(specM.getEncodings(), (encQ) => {
                 return isEnumSpec(encQ[prop]);
               });
+
+            case Property.SCALE_TYPE:
```

Comment:
Inspired from `encoding.ts` you can actually refactor to replace this `switch` with `if-else` and avoid having to add new property to the switch every time. 


Hint:  
- if mark
- leave // TODO transform there
- use `ENCODING_PROPERTIES` for second group
- `getNestedEncodingProperty` like how you suppose to use it in `encoding.ts`.   (Of course you have to figure that out first)